### DISCUSSION TITLE
How do you add recaptcha or prevent spamming?

### DISCUSSION THREAD

**Discussion by shielakn** [react-hook-form/react-hook-form#10529] [Q&A]
Hi! Does any know how to add recaptcha or prevent spamming? Thanks!

**Comment by craigmcc** [comment]
Googles recaptcha works well for me.  It even provides a nice react interface in npm package “react-google-recaptcha”.

https://www.google.com/recaptcha/about/

I use it in a few components AcceptForm. PasswordForm, and RegisterForm in a small app:

https://github.com/craigmcc/shoppinglist

The react app is in the “client” folder.
